src/Components/Endpoints/src/Discovery/ComponentInfo.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.AspNetCore.Components.Web;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Represents metadata about a component.
```

Comment:
Would be good to make this more descriptive, e.g., what kind of metadata, and what is it used for?